pkg/rbac/cluster_role.go

```diff
@@ -78,3 +78,15 @@ func DeleteClusterRole(c base.K8sClientWithLogging, ctx context.Context, cluster
 
 	return nil
 }
+
+// Validate ensures the cluster role does not contain cluster-admin permissions
+func (cr *ClusterRole) Validate() error {
```

Comment:
I think we only need to check the `cluster-admin` role in RoleBindings and ClusterRoleBindings.